internal/checker/check.go

```diff
@@ -62,7 +62,15 @@ func CheckRuleScript(importLoopData *shared.ImportLoopData, script string, scrip
 			Unsafe: false,
 		}
 		script = processScript(script, &ruleConf)
-		return ExecuteLuaMain(script, importLoopData, &ruleConf, scriptDirectory)
+		status, err := ExecuteLuaMain(script, importLoopData, &ruleConf, scriptDirectory)
+		if !status || err != nil {
+			return status, err
+		}
+		_, err = importLoopData.VRCT.Apply()
```

Comment:
This line concerns me since we only apply the VRCT if a user has confirmed to do so. And VRCT.Apply() is already called in cmd.

Reply:
Without this line, no changes are applied.   `api.fs.Apply` has been designed to apply VRCT changes before the end of execution.

Reply:
Yes, but the `ExecuteLuaMain` function is only supposed to return true if the rule passed requirements and false if it didn't. It mustn't  apply the changes. The changes are applied in cmd/check.go in the respective `check file` and `check` commands.